src/port.js

```diff
@@ -141,7 +141,22 @@ export default class Port {
 			if(req.id !== payload.id) {
 				continue;
 			}
-			req.promise(payload.val);
+
+			if (payload.hasOwnProperty('err')) {
+				const sentError = payload.err;
+
+				const e = new Error(sentError.message);
+				e.name = sentError.name;
+
+				for (let prop of Object.keys(sentError.props)) {
+					e[prop] = sentError.props[prop];
+				}
+
+				req.promise(Promise.reject(e));
```

Comment:
This error is hopefully _pretty close_ to the original. The stack is completely useless though.